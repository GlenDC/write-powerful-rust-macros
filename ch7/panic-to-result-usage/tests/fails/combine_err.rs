#![allow(unused_imports)]

use panic_to_result_macro::panic_to_result;
use std::convert::Infallible;

#[panic_to_result]
fn create_person_with_empty_panic(name: String, age: u32) -> Result<Person, Infallible> {
    if age > 30 {
        panic!();
    }
    Ok(Person { name, age })
}

fn main() {}
