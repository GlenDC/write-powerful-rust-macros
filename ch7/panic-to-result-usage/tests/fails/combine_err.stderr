error: cannot use macro on a function with Result as return type

         = help: try to remove the Result from the return type
         = note: we will wrap your return type in Result<_, String>

 --> tests/fails/combine_err.rs:7:62
  |
7 | fn create_person_with_empty_panic(name: String, age: u32) -> Result<Person, Infallible> {
  |                                                              ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: panic needs a message!

         = help: try to add a message: `panic!("message")`
         = note: we will add the message to Result's Err

 --> tests/fails/combine_err.rs:9:9
  |
9 |         panic!();
  |         ^^^^^^^^^
